src/main/scala/com/microsoft/sqlserver/jdbc/spark/DefaultSource.scala

```diff
@@ -52,7 +52,9 @@ class DefaultSource extends JdbcRelationProvider with Logging {
                     mode: SaveMode,
                     parameters: Map[String, String],
                     rawDf: DataFrame): BaseRelation = {
-        val options = new SQLServerBulkJdbcOptions(parameters)
+        // set SQL Server session application name to Spark MSSQL Connector + user input name
+        val applicationName = s"Spark MSSQL Connector ${parameters.getOrElse("applicationname", "")}"
+        val options = new SQLServerBulkJdbcOptions(parameters  + ("applicationname" -> applicationName))
```

Comment:
Is this sufficient or do we need to do the same change in each of the Strategy classes where the app name is set?

Reply:
each strategy is using option created in DefaultSource, which has already has applicationname, I think there is no need to add the applicationname in each strategy again.